src/services/useMoviesFetch.js

```diff
@@ -4,25 +4,31 @@ import { POPULAR_MOVIES_URL } from "../config";
 const useMoviesFetch = () => {
     const [state, setState] = useState({});
 
-    useEffect( async () => {
+    const fetchMovies = async (endpoint) => {
+        const page = endpoint.includes('page');
         try {
-            const resp = await fetch(POPULAR_MOVIES_URL);
+            const resp = await fetch(endpoint);
             const data = await resp.json();
-            setState({
-                movies: [...data.results],
-                heroBanner: data.results[0]
-            });
-            // setState(() => ({
-            //     movies: [...data.results],
-            //     heroBanner: data.results[0]
-            // }));
+            setState(prev => ({
+                movies:
+                page
+                ? [...prev.movies, ...data.results]
+                : [...data.results],
+                heroBanner: data.results[0],
+                currentPage: data.page,
+            }));
+
         }
         catch (e) {
             console.error(e);
         }
+    }
+
+    useEffect( () => {
+        fetchMovies(POPULAR_MOVIES_URL);
     }, []);
```

Comment:
fetch  refactoring:

- initial fetch of POPULAR_MOVIES_URL
- if button 'Load More' pressed, fetch passed URL (endpoint) and update the state.